test/nodetool/conftest.py

```diff
@@ -27,31 +27,51 @@ def pytest_addoption(parser):
                      " with --nodetool=cassandra, this should be the nodetool binary")
     parser.addoption('--jmx-path', action='store', default=None,
                      help="Path to the jmx binary, only used with --nodetool=cassandra")
+    parser.addoption('--run-within-unshare', action='store_true',
+                     help="Setup the 'lo' network if launched with unshare(1)")
 
 
 @pytest.fixture(scope="session")
-def rest_api_mock_server():
-    ip = f"127.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}"
-    port = random.randint(10000, 65535)
+def maybe_setup_loopback_network(request):
+    # unshare(1) -rn drops us in a new network namespace in which the "lo" is
+    # not up yet, so let's set it up first.
+    if request.config.getoption('--run-within-unshare'):
+        try:
+            args = "ip link set lo up".split()
```

Comment:
Two silly questions:
1. How come we have permissions to do this when not running under root?
2. Do we have "ip" in dbuild? If the answer is yes, why? :-)

Reply:
> Two silly questions:
> 
> 1. How come we have permissions to do this when not running under root?

we are root if we pass `-r` to `unshare` .

> 2. Do we have "ip" in dbuild? If the answer is yes, why? :-)

no. but i am under the impression that `ifconfig` is deprecated on some distros. for instance, Debian's base system does have `ifconfig` at all. despite that i don't develop scylla on Debian on daily basis, i still think it might be a good idea to support both tools.

Reply:
I see. Nice.